ServerCore/Pages/Events/Import.cshtml.cs

```diff
@@ -0,0 +1,155 @@
+
+        public ImportModel(PuzzleServerContext context)
+        {
+            _context = context;
+        }
+
+        [BindProperty]
+        public Event Event { get; set; }
+
+        public IList<Event> Events { get; set; }
+
+        [BindProperty]
+        public int ImportEventID { get; set; }
+
+        public async Task<IActionResult> OnGetAsync(int id)
+        {
+            Event = await _context.Events.SingleOrDefaultAsync(m => m.ID == id);
+
+            if (Event == null)
+            {
+                return NotFound();
+            }
+
+            Events = await _context.Events.Where(e => e != Event).ToListAsync();
+
+            return Page();
+        }
+
+        public async Task<IActionResult> OnPostImportAsync()
+        {
+            // the BindProperty only binds the event ID, let's get the rest
+            Event = await _context.Events.Where((e) => e.ID == Event.ID).FirstOrDefaultAsync();
+
+            if (Event == null || await _context.Events.Where((e) => e.ID == ImportEventID).FirstOrDefaultAsync() == null)
+            {
+                return NotFound();
+            }
+
+            // TODO: validate that you are an admin of both events!!!
+
+            var sourceEventAuthors = await _context.EventAuthors.Where((a) => a.Event.ID == ImportEventID).ToListAsync();
+            var sourcePuzzles = await _context.Puzzles.Where((p) => p.Event.ID == ImportEventID).ToListAsync();
+
+            // TODO: replace this with a checkbox and sufficient danger warnings about duplicate titles
+            bool deletePuzzleIfPresent = true;
+
+            using (var transaction = _context.Database.BeginTransaction())
+            {
+                // Step 1: Make sure all authors exist
+                foreach (var sourceEventAuthor in sourceEventAuthors)
+                {
+                    var destEventAuthor = await _context.EventAuthors.Where((e) => e.Event == Event && e.Author == sourceEventAuthor.Author).FirstOrDefaultAsync();
+
+                    if (destEventAuthor == null)
+                    {
+                        destEventAuthor = new EventAuthors(sourceEventAuthor);
+                        destEventAuthor.Event = Event;
+                        _context.EventAuthors.Add(destEventAuthor);
+                    }
+                }
+
+                // Step 2: Make sure all puzzles exist
+                Dictionary<int, Puzzle> puzzleCloneMap = new Dictionary<int, Puzzle>();
+
+                foreach (var sourcePuzzle in sourcePuzzles)
+                {
+                    // delete the puzzle if it exists
+                    if (deletePuzzleIfPresent)
+                    {
+                        foreach (Puzzle p in _context.Puzzles.Where((p) => p.Event == Event && p.Name == sourcePuzzle.Name))
+                        {
+                            // TODO why do I need a null check here? Does that mean something is improperly wired up?
```

Comment:
This might have been fixed in #144. I'm not sure if it's null if there are no contents